sourcegraph/sourcegraph
# Fix highlight "Code Search" navbar item

Closes https://github.com/sourcegraph/sourcegraph/issues/21373

- [x] Add integration or unit tests

## Test plan

Case 1 (search page):
- visit `/search?q=test&patternType=regexp`
- code search dropdown in the navigation `should be` highlighted


Case 2 (repo page):
- visit `/github.com/sourcegraph-testing/zap`
- code search dropdown in the navigation `should be` highlighted


Case 3 (repo file page):
- visit `/github.com/sourcegraph-testing/zap/-/blob/README.md`
- code search dropdown in the navigation `should be` highlighted


Case 3 (batch changes page):
- visit `/batch-changes`
- code search dropdown in the navigation `should not be` highlighted


## App preview:

- [Link](https://sg-web-naman-fix-highlighting-code.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-nriklnrukc.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
